### PR TITLE
Multiple Battery Support

### LINKED ISSUE #1288: Battery segment doesn't work with dual-battery laptops.
My laptop is Lenovo Thinkpad X240, and it got two batteries, which is `battery_BAT0` and `battery_BAT1` in UPower.
The battery segment of powerline display only the percentage of `battery_BAT0`, where it should be something like:
(`percentage of BAT0` * `capacity of BAT0` + `percentage of BAT1` * `capacity of BAT1`) / ( `capacity of BAT0` + `capacity of BAT1` )

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/8130381-battery-segment-doesn-t-work-with-dual-battery-laptops?utm_campaign=plugin&utm_content=tracker%2F283196&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F283196&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by mynameisfiber**
Adding multiple battery support to DBUS+UPOWER and /sys battery interfaces.  Fixes #1288 and hopefully mirrors #1522.

### CODE REVIEW

File: powerline/segments/common/bat.py [hunk 1]
```diff
@@ -46,39 +47,62 @@ def _fetch_battery_info(pl):
...
-					)
+								'State'
+							) != 2
+						return (energy * 100.0 / energy_full), state
+					return _flatten_battery
 				pl.debug('Not using DBUS+UPower as no batteries were found')
 
 	if os.path.isdir('/sys/class/power_supply'):
-		linux_capacity_fmt = '/sys/class/power_supply/{0}/capacity'
+		linux_energy_full_fmt = '/sys/class/power_supply/{0}/enery_full'
```
Comment (ZyX-I) [inline powerline/segments/common/bat.py]: @mynameisfiber Should not this be `energy`?

File: powerline/segments/common/bat.py [hunk 2]
```diff
@@ -30,6 +30,7 @@ def _fetch_battery_info(pl):
 			else:
 				devinterface = 'org.freedesktop.DBus.Properties'
 				devtype_name = interface + '.Device'
+				devices = []
```
Comment (ZyX-I) [inline powerline/segments/common/bat.py]: Have you tested your code with `import dbus` replaced with `raise ImportError`?

File: powerline/segments/common/bat.py [hunk 3]
```diff
@@ -30,6 +30,7 @@ def _fetch_battery_info(pl):
 			else:
```
Comment (ZyX-I) [inline powerline/segments/common/bat.py]: Have you tested your code with `import dbus` replaced with `raise ImportError`?

Comment (mynameisfiber) [inline powerline/segments/common/bat.py]: I just explicitly passed to the `/sys` method by settings `devices = None`

File: powerline/segments/common/bat.py (hunk 1 above)
Comment (mynameisfiber) [inline powerline/segments/common/bat.py]: No... we need to know the full capacity and the current capacity of each battery to properly weigh them in the average.

Comment (ZyX-I) [inline powerline/segments/common/bat.py]: @mynameisfiber This string contains substring `enery`. *Not* `energy`. This is what I mean.

Comment (mynameisfiber) [inline powerline/segments/common/bat.py]: Ah! You are completely right... I fixed the filenames in the most recent commit.

Comment (ZyX-I) [inline powerline/segments/common/bat.py]: PR was already merged. You need to run new PR.

Comment (ZyX-I) [inline powerline/segments/common/bat.py]: (This is also why I asked about testing: it should have failed with incorrect name.)

Comment (mynameisfiber) [inline powerline/segments/common/bat.py]: For sure... I ran the tests for this method before implementing the dbus method... must have been a stray replace-all? who knows... it works now.  New PR in order!

**Pull request merged**